tests/ConditionSqlTest.php

```diff
@@ -283,7 +283,7 @@ public function testArrayCondition()
 
         $m = new Model($this->db, 'user');
         $m->addField('name');
-        $m->addCondition('name', 'not', []); // this should not fail, always should be true
+        $m->addCondition('name', 'not in', []); // this should not fail, always should be true
```

Comment:
previous `not, []` is should be invalid, right, why did it not throw?

be very carefull and test what is/wanted behaviour of null - negate seems like to be not a negate and demorgan law can not be absolutely used then

Reply:
It did not throw as there was no check of operator validity when creating the condition. Now I have fixed that so tests failed here.

Reply:
perfect